packages/host/app/components/card-pill.gts

```diff
@@ -62,19 +63,43 @@ export default class CardPill extends Component<CardPillSignature> {
     return this.card && !this.card.id && !this.cardError;
   }
 
+  @action
+  private handleCardClick() {
+    if (this.args.isAutoAttachedCard && this.args.chooseCard) {
+      this.args.chooseCard(this.args.cardId);
+    }
+  }
+
+  @action
+  private handleRemoveClick(event: Event) {
+    // Prevent the click from bubbling up to the pill button
+    event.stopPropagation();
+    if (this.args.removeCard) {
+      this.args.removeCard(this.args.cardId);
+    }
+  }
+
+  private get pillKind() {
+    return this.args.isAutoAttachedCard && this.args.chooseCard
+      ? 'button'
+      : 'default';
+  }
+
```

Comment:
It seems like CardPill is getting too much specific knowledge about attached cards in it. Let's refactor to make it more of a UI component and not have isAutoAttachedCard / chooseCard. It should be told whether to use a dottedBorder and what to do onClick.